views/default/contests.html

```diff
@@ -131,19 +138,78 @@
                     'resource': event
                 });
 
-                request.execute(function(event) {
+                request.execute(function (event) {
                     $.web2py.flash("Reminder added!");
+                    checkContests();
                 });
             }
 
             /* Call Auth for Google Calendar */
-            handleAuthClick();
+            handleAuthorize(addEventToCalendar);
+        }
+
+        /* check contests using link and data/time */
+        function checkContests() {
+            $('#contests-table').find('> tbody > tr')
+                    /* only ongoing contests*/
+                    .filter(function (index, element) {
+                        return this.children.length === 6 && $('.stopstalk-timestamp', this).length === 1;
+                    })
+                    /* refer to https://developers.google.com/google-apps/calendar/v3/reference/events/list
+                     * for more info about constructing listing request
+                     */
+                    .each(function () {
+
+                        var contestName = this.children[0].textContent;
+                        var siteName = this.children[1].textContent;
+                        var startTime = this.children[2].textContent;
+                        // Note this will always be duration and not endtime
+                        var duration = this.children[3].textContent;
+                        var contestLink = this.children[4].firstChild.href;
+                        var calendarButton = $('.set-reminder', this);
+
+                        var startTimeStamp = new Date(startTime.replace(/-/g, '/'));
+                        var endTimeStamp = dateToTimeStamp(addMinutes(startTimeStamp, getMinutes(duration) - 5));
```

Comment:
https://contesttrackerapi.herokuapp.com/ gives timestamps in IST and in the present code adding to Google by IST timestamps (Timezone Asia/Kolkata).